### DISCUSSION TITLE
Request-FalconToken

### DISCUSSION THREAD

**Discussion by DiegoMessi1711** [CrowdStrike/psfalcon#169] [Q&A] [answered]
Am completely new to PSFalcon, just newly installed PSFalcon for a specific requirement. I have generated an API token on the Falcon UI and I am trying to request the falcon token, it results in this error: (Customer ID is redacted)

Write-Result : [{"code":403,"message":"Failed to generate access token for customer ID=xxxxxxxxxxxxxxxxxxxxxxxxxx"}]
At C:\Users\ebahukn\Documents\WindowsPowerShell\Modules\PSFalcon\2.1.6\Public\oauth2.ps1:131 char:27
+                 $Result = Write-Result $Response
+                           ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (System.Threadin...esponseMessage]:Task`1) [Write-Result], Exception
    + FullyQualifiedErrorId : 140ec5ca-f8e2-4882-b18c-e27b8f30e101,Write-Result

Am dropping an image of the Oath2.ps1, you could look at the line/chars referenced here. Am short of ideas as to what is wrong with this.  

![image](https://user-images.githubusercontent.com/98161553/150525150-662e4fcf-e0e7-4609-a9ea-b02a4e5f9ddd.png)

**Comment by bk-cs** [comment, marked as answer]
Most often, this error occurs when your token request is rejected due to attempting the request with the wrong cloud. Although the APIs are designed to redirect you to the right place, this doesn't work if your CID exists in `us-gov-1`, and sometimes fails due to PowerShell 5.1 handling the errors slightly differently than PowerShell Core (v7+).

Because `Request-FalconToken` defaults to the US-1 cloud (`https://api.crowdstrike.com`), you can resolve this by specifying the `-Cloud` or `-Hostname` parameter when running `Request-FalconToken`. 

```
Request-FalconToken -Cloud us-2
```
```
Request-FalconToken -Hostname https://api.us-2.crowdstrike.com
```

https://github.com/CrowdStrike/psfalcon/wiki/Authentication#alternate-clouds

**Comment by DiegoMessi1711** [comment]
This is already done. Could be closed.
